spotlight-client/src/AuthWall/AuthWall.tsx

```diff
@@ -0,0 +1,55 @@
+// Recidiviz - a data platform for criminal justice reform
+// Copyright (C) 2020 Recidiviz, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+import { useAuth0 } from "@auth0/auth0-react";
+import React from "react";
+import isAuthEnabled from "../utils/isAuthEnabled";
+
+/**
+ * Requires that the user is authenticated before rendering its children,
+ * and redirects unauthenticated users to an Auth0 login domain.
+ */
+const AuthChecker: React.FC = ({ children }) => {
+  const { isAuthenticated, isLoading, loginWithRedirect } = useAuth0();
+
+  if (isLoading) {
+    return <>Loading...</>;
```

Comment:
I know it doesn't matter _much_ because this is all for internal only, but I think it would be _nice_ to have a loading spinner at some point instead of just raw text. But don't mind me!

Reply:
yeah I am intending to replace this with something nicer once it exists